watertap/unit_models/electrocoagulation.py

```diff
@@ -0,0 +1,991 @@
+            description="Property package to use for control volume",
+            doc="""Property parameter object used to define property calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PhysicalParameterObject** - a PhysicalParameterBlock object.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "property_package_args",
+        ConfigBlock(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigBlock with arguments to be passed to a property block(s)
+    and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+
+    CONFIG.declare(
+        "electrode_material",
+        ConfigValue(
+            default="aluminum",
+            domain=In(ElectrodeMaterial),
+            description="Electrode material",
+        ),
+    )
+
+    CONFIG.declare(
+        "reactor_material",
+        ConfigValue(
+            default="stainless_steel",
+            domain=In(ReactorMaterial),
+            description="Reactor material",
+        ),
+    )
+
+    CONFIG.declare(
+        "overpotential_calculation",
+        ConfigValue(
+            default="fixed",
+            domain=In(OverpotentialCalculation),
+            description="Determination of overpotential",
+        ),
+    )
+
+    def build(self):
+
+        super().build()
+
+        solutes = self.config.property_package.solute_set
+
+        if "TDS" not in solutes:
+            raise ConfigurationError(
+                "TDS must be in feed stream for solution conductivity estimation."
+            )
+
+        self.scaling_factor = Suffix(direction=Suffix.EXPORT)
+
+        tmp_dict = dict(**self.config.property_package_args)
+        tmp_dict["has_phase_equilibrium"] = False
+        tmp_dict["parameters"] = self.config.property_package
+        tmp_dict["defined_state"] = True
+        self.properties_in = self.config.property_package.state_block_class(
+            self.flowsheet().config.time, doc="Material properties of inlet", **tmp_dict
+        )
+
+        tmp_dict["defined_state"] = False
+        self.properties_out = self.config.property_package.state_block_class(
+            self.flowsheet().config.time,
+            doc="Material properties of outlet",
+            **tmp_dict,
+        )
+
+        self.properties_waste = self.config.property_package.state_block_class(
+            self.flowsheet().config.time, doc="Material properties of waste", **tmp_dict
+        )
+
+        prop_in = self.properties_in[0]
+        prop_out = self.properties_out[0]
+        prop_waste = self.properties_waste[0]
+
+        self.add_port(name="inlet", block=self.properties_in)
+        self.add_port(name="outlet", block=self.properties_out)
+        self.add_port(name="waste", block=self.properties_waste)
```

Comment:
SHould this be termed "properties_waste"? What if you are recovering a resource as the "waste" stream? Would `properties_byproduct` be more general and not cause major problems for dependent analyses?

Reply:
Done